src/attr.rs

```diff
@@ -18,27 +21,39 @@ impl ToTokens for Display {
 }
 
 pub fn display(attrs: &[Attribute]) -> Result<Option<Display>> {
-    for attr in attrs {
-        if attr.path.is_ident("doc") {
-            let meta = attr.parse_meta()?;
-            let lit = match meta {
-                Meta::NameValue(syn::MetaNameValue {
-                    lit: syn::Lit::Str(lit),
-                    ..
-                }) => lit,
-                _ => unimplemented!(),
-            };
-
-            let lit = LitStr::new(lit.value().trim(), lit.span());
-
+    let attr = attrs
+        .iter()
+        .find(|attr| attr.path.is_ident("display"))
+        .or_else(|| attrs.iter().find(|attr| attr.path.is_ident("doc")));
+    if let Some(attr) = attr {
+        let meta = attr.parse_meta()?;
+        let lit = match meta {
+            Meta::NameValue(MetaNameValue { lit: Str(lit), .. }) => Some(lit),
+            Meta::NameValue(_) => unimplemented!("namevalue"),
+            Meta::Path(_) => unimplemented!("path"),
+            Meta::List(MetaList { nested, .. }) => {
+                nested.iter().find_map(|nested_attr| match nested_attr {
+                    NestedMeta::Meta(Meta::Path(path)) => {
+                        if path.is_ident("transparent") {
+                            Some(LitStr::new("{0}", attr.span()))
```

Comment:
So this is only partially complete. This covers the case where an error is implemented as a tuple, but thiserror also supports implying the source via a field named source on struct variants, and it supports manually marking a field as the source, so we need to add test cases for the following struct definitions:

```rust
struct Error {
    source: OtherError,
}

struct Error2 {
    #[source]
    cause: OtherError,
}
```

Reply:
I have added tests here to see the case you mentioned is working.
It's working.